src/CSI.IBTA.BenefitsService/Services/ClaimsService.cs

```diff
@@ -59,6 +59,34 @@ public async Task<GenericResponse<ClaimDto>> GetClaim(int claimId)
             return new GenericResponse<ClaimDto>(null, _mapper.Map<ClaimDto>(claim));
         }
 
+        public async Task<GenericResponse<bool>> UpdateClaim(int claimId, UpdateClaimDto updateClaimDto)
+        {
+            var claim = await _benefitsUnitOfWork.Claims
+                .Include(x => x.Enrollment)
+                .Include(x => x.Enrollment.Plan)
+                .Include(x => x.Enrollment.Plan.PlanType)
+                .Include(c => c.Enrollment.Plan.Package)
+                .FirstOrDefaultAsync(x => x.Id == claimId);
+
+            if (claim == null)
+                return new GenericResponse<bool>(HttpErrors.ResourceNotFound, false);
+
+            var enrollment = await _benefitsUnitOfWork.Enrollments
+                .Include(x => x.Plan)
+                .FirstOrDefaultAsync(x => x.EmployeeId == claim.Enrollment.EmployeeId && x.PlanId == updateClaimDto.PlanId);
+
+            if (enrollment == null)
+                return new GenericResponse<bool>(HttpErrors.ResourceNotFound, false);
+
+            claim.Enrollment = enrollment;
```

Comment:
Do we really need to update enrollment here? Sounds like we can override Enrollment for the Claim on demand. 

Lets imagine Employee enrolled on Plan 1 ( dental ) and Plan 2 ( cosmetic surgery). 
He claimed dentist visit on $20.0 on Plan 1 ( dental).  
UpdateClaim endpoint allows to change the claim amount ( this is OK, assuming that Admin can partially compensate the claim) and change enrollment of the claim to Plan 2 - and this is not OK, not clear what is the business sense of it.

Again, I can miss something, but double check - do we really need to change enrollment of a claim?

Reply:
In the wireframe, it shows that a claim's plan can be edited (and as a result it's part of this task's AC). Currently, in the database, the claim is directly linked with the enrollment. The only other option (if we want to be able to change claim's plan) would be to change the plan in the enrollment itself. 

This option didn't seem ideal to me because it changes the enrollment directly and the plan couldn't be changed back if employee didn't have multiple enrollments with the same plan (which I'm guessing would be the default situation).

By the way, claim editing is only available to pending claims (implemented in FE so maybe you didn't notice it), so, if I understood it correctly, the situation you described maybe wouldn't be as problematic if the relevant transactions haven't been made yet. 

But it's possible that perhaps I somehow misunderstood the requirement of changing the plan.